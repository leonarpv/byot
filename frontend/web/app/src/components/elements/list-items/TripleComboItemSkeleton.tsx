import * as React from 'react';
import {makeStyles, Theme} from '@material-ui/core';
import {WithStyles} from '@byot-frontend/web-common/src/types/WithStyles';
import {TripleComboItem} from './TripleComboItem';
import {PatchSkeleton} from '@byot-frontend/web-common/src/components/elementary/patch/PatchSkeleton';

interface Props extends WithStyles<typeof styles> {}

const styles = (theme: Theme) => ({
  title: {
    marginBottom: theme.spacing(1),
  },
  imagePlaceholder: {
    height: theme.spacing(8),
  },
  description: {},
});
const useStyles = makeStyles(styles);

export const TripleComboItemSkeleton: React.FC<Props> = (props: Props) => {
  const styles = useStyles(props);
  return (
    <TripleComboItem
      primary={<PatchSkeleton variant="rect" classes={{root: styles.title}} />}
      imagePlaceholder={<PatchSkeleton variant="rect" classes={{root: styles.imagePlaceholder}} />}
      description={<PatchSkeleton variant="rect" classes={{root: styles.description}} />}
    />
  );
};
