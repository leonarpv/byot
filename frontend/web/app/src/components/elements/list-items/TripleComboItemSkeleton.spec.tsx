import * as React from 'react';
import {render} from '@testing-library/react';
import {TripleComboItemSkeleton} from './TripleComboItemSkeleton';

describe('<TripleComboItemSkeleton/>', () => {
  it('should render', () => {
    const {container} = render(<TripleComboItemSkeleton />);
    expect(container).toMatchSnapshot();
  });
});
