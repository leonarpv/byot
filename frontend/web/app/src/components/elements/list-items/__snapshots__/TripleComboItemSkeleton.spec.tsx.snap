// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TripleComboItemSkeleton/> should render 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-12 MuiListItem-root makeStyles-root-4 MuiListItem-gutters MuiListItem-alignItemsFlexStart makeStyles-root-11"
  >
    <div
      class="makeStyles-image-5 makeStyles-image-8"
    >
      <span
        class="MuiSkeleton-root makeStyles-root-13 makeStyles-imagePlaceholder-2 MuiSkeleton-rect MuiSkeleton-pulse"
      />
    </div>
    <div
      class="MuiListItemText-root makeStyles-textRoot-7 makeStyles-textRoot-10 MuiListItemText-multiline"
    >
      <div
        class="MuiTypography-root MuiListItemText-primary makeStyles-primaryText-6 makeStyles-primaryText-9 MuiTypography-body1 MuiTypography-displayBlock"
      >
        <span
          class="MuiSkeleton-root makeStyles-root-13 makeStyles-title-1 MuiSkeleton-rect MuiSkeleton-pulse"
        />
      </div>
      <div
        class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        <span
          class="MuiSkeleton-root makeStyles-root-13 makeStyles-description-3 MuiSkeleton-rect MuiSkeleton-pulse"
        />
      </div>
    </div>
  </div>
</div>
`;
